chain/chunks/src/client.rs

```diff
@@ -0,0 +1,13 @@
+use actix::Message;
+use near_network::types::MsgRecipient;
+use near_primitives::sharding::ShardChunkHeader;
+
+#[derive(Message)]
+#[rtype(result = "()")]
+pub enum ShardsManagerResponse {
+    ChunkCompleted(ShardChunkHeader),
+}
+
+pub trait ClientAdapterForShardsManager: MsgRecipient<ShardsManagerResponse> {}
```

Comment:
Let's maybe hide actix from the API here?

```diff
diff --git a/chain/chunks/src/client.rs b/chain/chunks/src/client.rs
index ed2b2ba58..e207c0cd1 100644
--- a/chain/chunks/src/client.rs
+++ b/chain/chunks/src/client.rs
@@ -2,12 +2,18 @@ use actix::Message;
 use near_network::types::MsgRecipient;
 use near_primitives::sharding::ShardChunkHeader;
 
+pub trait ClientAdapterForShardsManager {
+    fn did_complete_chunk(&self, chunk_header: ShardChunkHeader);
+}
+
 #[derive(Message)]
 #[rtype(result = "()")]
 pub enum ShardsManagerResponse {
     ChunkCompleted(ShardChunkHeader),
 }
 
-pub trait ClientAdapterForShardsManager: MsgRecipient<ShardsManagerResponse> {}
-
-impl<A: MsgRecipient<ShardsManagerResponse>> ClientAdapterForShardsManager for A {}
+impl<A: MsgRecipient<ShardsManagerResponse>> ClientAdapterForShardsManager for A {
+    fn did_complete_chunk(&self, chunk_header: ShardChunkHeader) {
+        self.do_send(ShardsManagerResponse::ChunkCompleted(chunk_header))
+    }
+}
diff --git a/chain/chunks/src/lib.rs b/chain/chunks/src/lib.rs
index 990d7cbee..b376f353d 100644
--- a/chain/chunks/src/lib.rs
+++ b/chain/chunks/src/lib.rs
@@ -85,7 +85,7 @@ use std::sync::Arc;
 use std::time::{Duration, Instant};
 
 use chrono::DateTime;
-use client::{ClientAdapterForShardsManager, ShardsManagerResponse};
+use client::ClientAdapterForShardsManager;
 use near_primitives::time::Utc;
 use rand::seq::IteratorRandom;
 use rand::seq::SliceRandom;
@@ -1967,7 +1967,7 @@ impl ShardsManager {
         self.encoded_chunks.remove_from_cache_if_outside_horizon(chunk_hash);
         self.requested_partial_encoded_chunks.remove(chunk_hash);
         debug!(target: "chunks", "Completed chunk {:?}", chunk_hash);
-        self.client_adapter.do_send(ShardsManagerResponse::ChunkCompleted(header));
+        self.client_adapter.did_complete_chunk(header);
     }
 
     /// Send the parts of the partial_encoded_chunk that are owned by `self.me` to the
```

In generally we want to move away from it, so better if stuff doesn't call actix APIs directly

Reply:
Thanks @matklad; @mzhangmzz, is this the same thing you were talking about? It doesn't solve the fact that this will go on the same client actor queue though. I don't see how to avoid that...

Reply:
Let me do this for the next PR; I need to change the API anyway.